src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -1385,6 +1389,41 @@ namespace
         }
         else
         {
+            asr::AOVContainer aovs;
+
+            IRenderElementMgr* re_manager = GetCOREInterface()->GetCurRenderElementMgr();
+            re_manager->SetDisplayElements(false);
+            for (size_t i = 0, e = re_manager->NumRenderElements(); i < e; ++i)
+            {
+                const asr::AOVFactoryRegistrar factory_registrar;
+                auto factories = factory_registrar.get_factories();
+
+                auto* render_element = re_manager->GetRenderElement(static_cast<int>(i));
+                render_element->SetEnabled(false);
+
+                if (render_element->ClassID() == AppleseedRenderElement::get_class_id())
+                {
+                    AppleseedRenderElement* element = static_cast<AppleseedRenderElement*>(render_element);
+                    int aov_index = 0;
+                    element->GetParamBlock(0)->GetValueByName(L"aov_index", 0, aov_index, FOREVER);
+                    if (aov_index > 0 && aov_index <= factories.size())
```

Comment:
Signed vs unsigned comparison here. I'm surprised you're not getting a warning. You will probably need to cast `factories.size()` to `int`.